### PR TITLE
Unify training

### PR DESCRIPTION
**Pull request by ericcanton**
* removes class variables from the MO classifier
* adds `**training_kwargs` to the training function, auto set up class weights.
    * this _should_ have the same effect as simply not selecting those array indices, for the simpler classifiers this worked with before, but treat the whole thing uniformly. Shout-out to sklearn AND tensorflow for having all of their classifiers accept `sample_weight` in the `.fit()` method. And to you, @ftavella , for all you did to unify things before. It really sped this up!

### CODE REVIEW

File: pisces/models.py [hunk 1]
```diff
@@ -74,35 +74,58 @@ def predict_probabilities(self, sample_X: np.ndarray | pl.DataFrame) -> np.ndarr
...
     Xs = np.concatenate(examples_X, axis=0)
     ys = np.concatenate(examples_y, axis=0)
 
     selector = ys >= 0
-    Xs = Xs[selector]
-    ys = ys[selector]
+
+    # adds "model__sample_weight" to train_kwargs, but if it already exists, it will not overwrite it
+    # the order of the dictionaries in | important, as the rightmost dictionary will overwrite the leftmost
+    train_kwargs = {classifier.model_pipeline_name + '__sample_weight': selector} \
```
Comment (ftavella) [inline pisces/models.py:122]: Nice! Didn't know about this trick

File: pisces/models.py [hunk 2]
```diff
@@ -74,35 +74,58 @@ def predict_probabilities(self, sample_X: np.ndarray | pl.DataFrame) -> np.ndarr
...
+        | train_kwargs
 
     loss_list = []
     old_stdout = sys.stdout
     sys.stdout = mystdout = StringIO()
-    classifier.pipeline.fit(Xs, ys) # Fit the model
+    # multiply ys by selector, to zero-out the "-1" masked values but leave the others unchanged (where selector == 1)
```
Comment (ftavella) [inline pisces/models.py:128]: This is to avoid any errors caused by a negative value on `ys` right? Because those samples won't count towards the loss since we have set their `sample_weight` to 0 on `selector`

File: pisces/models.py [hunk 3]
```diff
@@ -205,6 +231,22 @@ def __init__(
             model=tf_model,
```
Comment (ftavella) [inline pisces/models.py:240]: Should we call this init with `scaler_pipeline_name='None'` to make clear that we are not using a scaler? Because if not by default this attribute will be set to `'scaler'`

**Review by ftavella — APPROVED** [review]
Just left a couple of comments but the rest looks good!
